paseos/attitude/attitude_model.py

```diff
@@ -0,0 +1,229 @@
+
+        Args:
+            dt (float): time to advance
+
+        Returns: rotation vector of spacecraft body expressed in the body frame
+        """
+        # todo: check if need to skip first step
+        # theta_2:
+
+        # to not have the spacecraft rotate in the first timestep:
+        # if self._first_run:
+        #    body_rotation = self._actor_angular_velocity * dt
+        #    self._actor_theta_2 = body_to_rpy(body_rotation, self._actor_attitude_in_rad)
+        #    self._first_run = False
+        self.calculate_angular_acceleration()
+        self._actor_angular_velocity += self._actor_angular_acceleration * dt
+        body_rotation = self._actor_angular_velocity * dt
+        return body_to_rpy(body_rotation, self._actor_attitude_in_rad)
+
+    def frame_rotation(self, position, previous_position, velocity):
+        """Calculate the rotation vector of the RPY frame rotation within the inertial frame.
+        this rotation component makes the actor body attitude stay constant w.r.t. inertial frame,
+
+        Args:
+            position (np.ndarray): actor position vector
+            previous_position (np.ndarray): actor position vector in previous timestep
+            velocity (np.ndarray): actor velocity vector
+
+        Returns: rotation vector of RPY frame w.r.t. ECI frame expressed in the ECI frame
+        """
+        # orbital plane normal unit vector: (p x v)/||p x v||
+        orbital_plane_normal = np.cross(position, velocity) / np.linalg.norm(
+            np.cross(position, velocity)
+        )
+
+        # rotation angle: arccos((p . p_previous) / (||p|| ||p_previous||))
+        rpy_frame_rotation_angle_in_eci = np.arccos(
+            np.linalg.multi_dot([position, previous_position])
+            / (np.linalg.norm(position) * np.linalg.norm(previous_position))
+        )
+
+        # assign this scalar rotation angle to the vector perpendicular to rotation plane
+        rpy_frame_rotation_vector_in_eci = (
+            orbital_plane_normal * rpy_frame_rotation_angle_in_eci
+        )
+
+        # this rotation needs to be compensated in the rotation of the body frame, so it's attitude stays fixed
+        return -eci_to_rpy(rpy_frame_rotation_vector_in_eci, position, velocity)
+
+    def body_axes_in_rpy(self):
+        """Transforms vectors expressed in the spacecraft body frame to the roll pitch yaw frame.
+        Vectors: - x, y, z axes
+                 - user specified pointing vector
+
+        Returns: transformed vectors
+        """
+        # principal axes:
+        x = body_to_rpy(np.array([1, 0, 0]), self._actor_attitude_in_rad)
+        y = body_to_rpy(np.array([0, 1, 0]), self._actor_attitude_in_rad)
+        z = body_to_rpy(np.array([0, 0, 1]), self._actor_attitude_in_rad)
+
+        # pointing vector:
+        p = body_to_rpy(self._actor_pointing_vector_body, self._actor_attitude_in_rad)
+        return x, y, z, p
+
+    def update_attitude(self, dt):
```

Comment:
Make sure this function is always called inside PASEOS main loop to be sure the variables will be updated all at the same time: https://github.com/aidotse/PASEOS/blob/e5bb071b4ee2988c2765497aa8fe3d0f6eacb655/paseos/paseos.py#L123